packages/modules/vault/src/vault-container.test.ts

```diff
@@ -0,0 +1,54 @@
+import vault from "node-vault";
+import { setTimeout } from "node:timers/promises";
+import { StartedVaultContainer, VaultContainer } from "./vault-container";
+
+const VAULT_TOKEN = "my-root-token";
+
+describe("vault", { timeout: 180_000 }, () => {
+  let container: StartedVaultContainer;
+
+  afterEach(async () => {
+    await container?.stop();
+  });
```

Comment:
The tests are passing but there is a race condition. The tests run concurrently by default, so it's possible one test will stop the other's container (and likely other tests' containers aren't being stopped). You can reproduce this by wrapping a test in a for loop, you'll see it starts to fail.

In the future we'll be able to use TS 5.2's `using` to get rid of this boilerplate, but for now we need it.

Reply:
Thanks for the heads up!

https://github.com/testcontainers/testcontainers-node/pull/1079/commits/05211d132ea2d40fb8493dca0ea06517ca257c97